pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationGroupByOrderByPlanNode.java

```diff
@@ -58,6 +63,14 @@ public AggregationGroupByOrderByPlanNode(IndexSegment indexSegment, QueryContext
     List<ExpressionContext> groupByExpressions = queryContext.getGroupByExpressions();
     assert groupByExpressions != null;
     _groupByExpressions = groupByExpressions.toArray(new ExpressionContext[0]);
+    _queryContext = queryContext;
+
+    // Only trim if there is OrderBy and has a positive trim size
+    if (queryContext.getOrderByExpressions() != null && minSegmentTrimSize > 0) {
```

Comment:
I feel we might want to pass `minSegmentTrimSize` to the operator, and the operator can decide whether to trim it or not

Reply:
Any reason for this behavior rather than figuring out in the plan?

Reply:
Because we have to do the actual comparison in the operator (during query run time), so imo more clear if we group the comparison in one place

Reply:
I see